more_itertools/more.py

```diff
@@ -1121,12 +1132,17 @@ def sliced(seq, n):
         >>> list(sliced((1, 2, 3, 4, 5, 6, 7, 8), 3))
         [(1, 2, 3), (4, 5, 6), (7, 8)]
 
+    `sliced` raises a `ValueError` if *strict* is `True` and the length of
+    the sequence is not divisible by the requested slice length.
+
     This function will only work for iterables that support slicing.
     For non-sliceable iterables, see :func:`chunked`.
 
     """
-    return takewhile(len, (seq[i : i + n] for i in count(0, n)))
```

Comment:
Could we make this match the pattern in `chunked`? If `strict` is `False`, do nothing different. Otherwise, do the extra checking.